packages/invoices/src/OutboundShipment/DetailView/modals/hooks/useNextItem.test.tsx

```diff
@@ -0,0 +1,182 @@
+import { useNextItem } from './useNextItem';
+
+const getInvoice = () => ({
+  __typename: 'InvoiceNode',
+  id: '',
+  lines: {
+    __typename: 'InvoiceLineConnector',
+    nodes: [] as InvoiceLineNode[],
+    totalCount: 0,
+  },
+  otherParty: {
+    __typename: 'NameNode',
+    id: '',
+  },
+  pricing: {
+    __typename: 'InvoicePricingNode',
+  },
+});
+
+const getLines = (): InvoiceLineNode[] => [
+  {
+    id: 'a',
+    itemId: 'a',
+    sellPricePerPack: 0,
+    type: InvoiceLineNodeType.StockOut,
+    __typename: 'InvoiceLineNode',
+    costPricePerPack: 0,
+    invoiceId: '',
+    itemCode: '',
+    itemName: '',
+    numberOfPacks: 0,
+    packSize: 1,
+    item: {
+      id: 'a',
+      __typename: 'ItemNode',
+      availableBatches: {
+        __typename: 'StockLineConnector',
+        nodes: [],
+        totalCount: 0,
+      },
+      code: '',
+      isVisible: true,
+      name: '',
+    },
+  },
+  {
+    id: 'b',
+    itemId: 'b',
+    sellPricePerPack: 0,
+    type: InvoiceLineNodeType.StockOut,
+    __typename: 'InvoiceLineNode',
+    costPricePerPack: 0,
+    invoiceId: '',
+    itemCode: '',
+    itemName: '',
+    numberOfPacks: 0,
+    packSize: 1,
+    item: {
+      id: 'b',
+      __typename: 'ItemNode',
+      availableBatches: {
+        __typename: 'StockLineConnector',
+        nodes: [],
+        totalCount: 0,
+      },
+      code: '',
+      isVisible: true,
+      name: '',
+    },
+  },
+  {
+    id: 'c',
+    itemId: 'c',
+    sellPricePerPack: 0,
+    type: InvoiceLineNodeType.StockOut,
+    __typename: 'InvoiceLineNode',
+    costPricePerPack: 0,
+    invoiceId: '',
+    itemCode: '',
+    itemName: '',
+    numberOfPacks: 0,
+    packSize: 1,
+    item: {
+      id: 'c',
+      __typename: 'ItemNode',
+      availableBatches: {
+        __typename: 'StockLineConnector',
+        nodes: [],
+        totalCount: 0,
+      },
+      code: '',
+      isVisible: true,
+      name: '',
+    },
+  },
+];
+
+describe('useNextItem', () => {
+  const server = setupServer();
+
+  beforeAll(() => {
+    server.listen();
+  });
+
+  afterEach(() => {
+    server.resetHandlers();
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it('eventually equals an object where the next item is equal to the next item in sorted order.', () => {
+    const invoice = getInvoice();
+    invoice.lines.nodes = getLines();
+    const handler = graphql.query('invoice', (_, res, ctx) => {
+      return res(ctx.data({ invoice }));
+    });
+    server.use(handler);
+    const result = renderHook(() => useNextItem('a'), {
+      wrapper: TestingProvider,
+    });
+
+    return result.waitForNextUpdate().then(() => {
+      expect(result.result.current.next).toEqual(
+        expect.objectContaining({ id: invoice.lines.nodes[1]?.id })
+      );
+      expect(result.result.current.disabled).toEqual(false);
+    });
+  });
+
+  it('eventually equals an object where if the next item is the last, disabled is true', async () => {
```

Comment:
I think it should be `disabled` if this is the last item. not sure why it should be disabled if the next item is the last one?

Reply:
uhh.. wow! What a foooool! Fixed ! 😁